locallib.php

```diff
@@ -422,3 +446,42 @@ function local_boostnavigation_user_has_role_on_page($userid, $setting) {
         return count(array_intersect($rolesincontextshortnames, $showforroles)) > 0;
     }
 }
+
+/**
+ * Checks if the user has any of the allowed global system roles.
+ * @param int $userid
+ * @param string $setting A comma-separated whitelist of allowed system roles.
+ * @return bool
+ */
+function local_boostnavigation_user_has_role_on_system($userid, $setting) {
+
+    $hasrole = null;
+
+    // Split optional setting by comma.
+    $showforroles = explode(',', $setting);
+
+    // Get roles for user.
+    $roles = get_user_roles(context_system::instance(), $userid, false);
+
+    foreach ($roles as $role) {
+        if (in_array($role->shortname, $showforroles)) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+/**
+ * Checks if the user is a site-admin and the parameter is set true.
+ * @param int $userid
+ * @param string $setting A value which describes the state, e.g. true.
+ * @return bool
+ */
+function local_boostnavigation_user_is_admin($userid, $setting) {
+    if ($setting == "true") {
+        return is_siteadmin($userid);
+    } else {
+        return false;
+    }
+}
```

Comment:
It would be nice if there would still be an empty line at the end of the file as it is best practice in Moodle development.